.github/workflows/release.yml

```
@@ -1,40 +1,122 @@
 ---
 name: Release
 
-on:
-  push:
-    tags:
-      - '*'
+on: push
+
+env:
+  # Change these for your project's URLs
+  PYPI_URL: https://pypi.org/p/django-simple-history
+  PYPI_TEST_URL: https://test.pypi.org/p/django-simple-history
 
 jobs:
+
   build:
-    if: github.repository == 'jazzband/django-simple-history'
+    name: Build distribution 📦
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v4
-        with:
-          fetch-depth: 0
-
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
-          python-version: 3.x
+          python-version: "3.x"
+      - name: Install pypa/build
+        run:
+          python3 -m pip install build --user
+      - name: Build a binary wheel and a source tarball
+        run: python3 -m build
+      - name: Store the distribution packages
+        uses: actions/upload-artifact@v4
+        with:
+          name: python-package-distributions
+          path: dist/
+
+  publish-to-pypi:
+    name: >-
+      Publish Python 🐍 distribution 📦 to PyPI
+    if: startsWith(github.ref, 'refs/tags/')  # only publish to PyPI on tag pushes
+    needs:
+      - build
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: ${{ env.PYPI_URL }}
+    permissions:
+      id-token: write  # IMPORTANT: mandatory for trusted publishing
+    steps:
+      - name: Download all the dists
+        uses: actions/download-artifact@v4
+        with:
+          name: python-package-distributions
+          path: dist/
+      - name: Publish distribution 📦 to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1.10
```

I'm not familiar with the `release/` prefix to the version number - what's the function of that? Does it make any semantic difference? 🤔

We should probably also update to v1.12, which, to me, doesn't seem like it should require any other changes 🙂
```suggestion
        uses: pypa/gh-action-pypi-publish@v1.12
```
The same goes for TestPyPI below.